client/scala/src/main/scala/ai/verta/repository/Commit.scala

```diff
@@ -0,0 +1,16 @@
+package ai.verta.repository
+
+import ai.verta.swagger.client.ClientSet
+import ai.verta.swagger._public.modeldb.versioning.model._
+
+import scala.concurrent.ExecutionContext
+import scala.util.{Failure, Success, Try}
+
+/** Commit within a ModelDB Repository
+ *  There should not be a need to instantiate this class directly; please use Repository.getCommit methods
+ */
+class Commit(
+  val clientSet: ClientSet, val repo: VersioningRepository,
+  var commit: VersioningCommit, var commit_branch: Option[String] = None
```

Comment:
Avoid exposing `var`s. Also, these are not being modified, so no need for `var` right now. Also, make sure the naming convention is maintained. Don't mix snake case and camel case.

Reply:
Ah yeah I used var here originally because the bulk of commit operations require modifying the commit instance in-place. I have changed it to val for now, but we might need to change this in future PRs. Resolved.